DcNotificationService/NotificationService.swift

```diff
@@ -14,21 +14,36 @@ class NotificationService: UNNotificationServiceExtension {
             contentHandler(silenceNotification())
             return
         }
-        UserDefaults.setNseFetching()
+        UserDefaults.setNseFetching(until: Date().addingTimeInterval(26))
 
         // as we're mixing in notifications from accounts without PUSH and we cannot add multiple notifications,
         // it is best to move everything to the same thread - and set just no threadIdentifier
 
         dcAccounts.openDatabase(writeable: false)
         let eventEmitter = dcAccounts.getEventEmitter()
 
-        if !dcAccounts.backgroundFetch(timeout: 25) {
-            UserDefaults.pushToDebugArray("ERR3")
-            UserDefaults.setNseFetching(false)
+        // Send the bestAttempt notification when memory is critical because the process will be killed
+        // by the system soon and any notification is better than nothing.
+        var exitedDueToCriticalMemory = false
+        let memoryPressureSource = DispatchSource.makeMemoryPressureSource(eventMask: .critical)
+        memoryPressureSource.setEventHandler {
+            // Order of importance because we might crash very soon
+            // TODO: Could notify with "received message too big to decrypt in background"
             contentHandler(bestAttemptContent)
+            exitedDueToCriticalMemory = true
+            UserDefaults.pushToDebugArray("ERR5")
+        }
+        memoryPressureSource.activate()
+
+        guard dcAccounts.backgroundFetch(timeout: 25) && !exitedDueToCriticalMemory else {
+            UserDefaults.pushToDebugArray("ERR3")
+            UserDefaults.setNseFetching(until: nil)
+            if !exitedDueToCriticalMemory {
+                contentHandler(bestAttemptContent)
+            }
```

Comment:
there are two more calls to contentHandler() below that we may want to protect the same way

(if we want to do so, i would just repeat 

```
            if !exitedDueToCriticalMemory {
                contentHandler(bestAttemptContent)
            }
``` 

there, factoring out to a function or so may lower understandability of the code, kiss :)

no blocker, tho

Reply:
it will never get there if it exits due to critical memory (assuming this exit happens after the fetch background, which I think is safe because that should clean up after itself)